File: programming tasks/855 - Lunch in Grid City/855.py
import sys
import math

lines = sys.stdin.readlines()
numberOfTestCases = int(lines[0])
lines.pop(0)

for i in range(numberOfTestCases):
    numberOfPeople= int(lines[0].split(" ", 2)[2])

    lines.pop(0)
    xcoords = list()
    ycoords = list()
    for j in range(numberOfPeople):
        line = lines.pop(0)
        pos = line.split(" ",1)
        xcoords.append(int(pos[0]))
        ycoords.append(int(pos[1]))

    xcoords.sort()
    ycoords.sort()
    street = xcoords[math.ceil(numberOfPeople/2-1)]
    avenue = ycoords[math.ceil(numberOfPeople/2-1)]
    print("(Street: "+ str(street)+ ", Avenue: "+ str(avenue)+ ")")

File: programming tasks/855 - Lunch in Grid City/855.cpp
#include <iostream>
#include <string>
#include <algorithm>
using namespace std;
int main()
{
    int anzahlprobleme, xsize, ysize, people;
    cin >> anzahlprobleme;
    
    for(int i = 0; i < anzahlprobleme; i++){
        cin >> xsize >> ysize >> people;
        
        int x [people];
        int y [people];
        
        for(int j = 0; j < people; j++){
            cin >> x[j] >> y[j];
        }
        
        sort(x,x + people);
        sort(y,y + people);
        
        cout << "(Street: " << x[(people-1)/2] << ", Avenue: " <<  y[(people-1)/2] << ")\n";
        
        
    }
    
}